test/e2e/apps/job.go

```diff
@@ -344,6 +344,61 @@ var _ = SIGDescribe("Job", func() {
 		}
 	})
 
+	ginkgo.It("should recreate pods only after they have failed if pod replacement policy is set to Failed", func(ctx context.Context) {
+		ginkgo.By("Creating a job")
+		job := e2ejob.NewTestJob("", "pod-recreate-failed", v1.RestartPolicyNever, 1, 1, nil, 1)
+		job.Spec.PodReplacementPolicy = ptr.To(batchv1.Failed)
+		job.Spec.Template.Spec.Containers[0].Command = []string{"/bin/sh", "-c", `_term(){
+	sleep 5
```

Comment:
Is this necessary if we follow the design proposed here: https://github.com/kubernetes/kubernetes/pull/121491#discussion_r1372003113? I think we could use the "notTerminate" image, and  another call for `DeletePodsWithGracePeriod` with `0`. Then, we don't need the 5s penalty.